### PR TITLE
Add --grep option to run a subset of test files

### PR DESCRIPTION
**Pull request by bearfriend**
[US152300](https://rally1.rallydev.com/#/?detail=/userstory/699970845705&fdp=true): Visual-diff: run a subset of tests

Allows any library-created `group` to run a subset of test files by modifying the `pattern` with values from the `--grep` (`-g`) CLI arg.

`npm run test:visual-diff -- -g button list` would modify the pattern `./test/**/*.vdiff.js` to `['./test/**/*button*.vdiff.js', './test/**/*list*.vdiff.js']`.

### CODE REVIEW

File: src/server/wtr-config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { argv } from 'node:process';
 import { defaultReporter } from '@web/test-runner';
 import { playwrightLauncher } from '@web/test-runner-playwright';
+import { readCliArgs } from '../../node_modules/@web/test-runner/dist/config/readCliArgs.js';
```
Comment (bearfriend) [inline src/server/wtr-config.js:3]: We likely can use a generic library to get these with less risk.

File: src/server/wtr-config.js [hunk 2]
```diff
@@ -62,15 +62,21 @@ export class WTRConfig {
...
 	create({
 		pattern = DEFAULT_PATTERN,
 		vdiff = DEFAULT_VDIFF,
 		timeout,
 		...passthroughConfig
 	} = {}) {
 
-		if (!this.#cliArgs.includes('--group') || this.#cliArgs.includes('default')) {
-			if (this.#cliArgs.includes('--playwright')) {
+		if (!this.#cliArgs.group?.length || this.#cliArgs.group?.includes('default')) {
```
Comment (bearfriend) [inline src/server/wtr-config.js:78]: Not an array
```suggestion
		if (!this.#cliArgs.group || this.#cliArgs.group === 'default') {
```

File: src/server/wtr-config.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
-import { argv } from 'node:process';
+import commandLineArgs from 'command-line-args';
```
Comment (dlockhart) [inline src/server/wtr-config.js:1]: Will want to add an explicit dependency on this.

**Comment by bearfriend** [comment]
> @eKoopmans, do you remember what syntax you were using with WTR to do this out-of-thebox? I think it was `npx wtr ./**/icon.vdiff.js` but that isn't working for me.

This looks liken it's taking advantage of `--files` being the default option, which, by itself, doesn't work if the group already has a `file` config, and all ours do. In my first commit I used `--files` to do what I'm doing now with `--grep`, but I didn't like mixing the meaning depending on whether you're using a library-defined group or your own group.

**Comment by eKoopmans** [comment]
> @eKoopmans, do you remember what syntax you were using with WTR to do this out-of-thebox? I think it was `npx wtr ./**/icon.vdiff.js` but that isn't working for me.

Okay I haven't looked at this PR so I have no other context, but to answer your question this is the sort of pattern I use:

```
npx web-test-runner **/object-property-list*.test.js --group default
```

File: test/server/wtr-config.test.js [hunk 4]
```diff
@@ -97,6 +97,13 @@ describe('createWtrConfig', () => {
 			expect(config.groups[0]).to.not.have.property('name', 'vdiff');
 		});
 
+		it('should filter test files using --files values', () => {
```
Comment (bearfriend) [inline test/server/wtr-config.test.js:100]:
```suggestion
		it('should filter test files using --grep values', () => {
```

**Review by dlockhart — APPROVED** [review]
Definitely think this covers the use case! 🎉

**Pull request merged**